internal/artifactory_cache_test.go

```diff
@@ -0,0 +1,139 @@
+package internal
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/cisco-open/grabit/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestAddWithArtifactoryCache verifies adding a resource with caching enabled.
+func TestAddWithArtifactoryCache(t *testing.T) {
+	// Sub-test to verify behavior when the token is not set.
+	t.Run("TokenNotSet", func(t *testing.T) {
+		// Clear the GRABIT_ARTIFACTORY_TOKEN environment variable.
+		t.Setenv("GRABIT_ARTIFACTORY_TOKEN", "")
+
+		// Setup a simple HTTP handler that always returns "test content".
```

Comment:
Is all this setup really necessary if you're only testing the case where GRABIT_ARTIFACTORY_TOKEN isn't set?

Reply:
@rabadin Looks like without defining "t.Setenv("GRABIT_ARTIFACTORY_TOKEN", "")" the test fails when I run "make test" but passes when I run the test directly. Do you have any recommendations for this? 

Went ahead and pushed this change to see if it passes the CI.

Reply:
I'm not sure what you mean by "when I run the test directly". Do you mean running this individual test in isolation or manually testing things?
I see that the tests pass right but you're not testing all the functionality, which is reflected in the new coverage number (with this PR): 75.8%.

Reply:
@rabadin I mean when I run the test in isolation. 

I think one way to up the testing coverage could be testing the normal Artifactory download functionality, I don't think this is currently being tested. If you think this would be a good test to add I can work on adding it when I get home this evening.

Reply:
By definition the UT coverage is the coverage of the UT.

Reply:
Well, yes, you clearly need to test:
a) the happy path
- Adding a resources with a cache
- Downloading a resource with a cache, getting the data from the cache (using mocking, as discussed)
- Deleting a resource with a cache, deleting the cache (using mocking, as discussed)
b) the non happy path
- Adding resource with a cache but no token env var
- Downloading a resource with a cache, but the cache GET errors
- Deleting a resource with a cache, and deleting the cache works